src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -48,28 +51,40 @@ public class ConsoleRunner
 
         public ConsoleRunner(ITestEngine engine, ConsoleOptions options, ExtendedTextWriter writer)
         {
-            _engine = engine;
-            _options = options;
-            _outWriter = writer;
-
-            _workDirectory = options.WorkDirectory ?? Directory.GetCurrentDirectory();
-
-            if (!Directory.Exists(_workDirectory))
-                Directory.CreateDirectory(_workDirectory);
+            Guard.ArgumentNotNull(_engine = engine, nameof(engine));
+            Guard.ArgumentNotNull(_options = options, nameof(options));
+            Guard.ArgumentNotNull(_outWriter = writer, nameof(writer));
 
             _resultService = _engine.Services.GetService<IResultService>();
+            Guard.OperationValid(_resultService != null, "Internal Error: ResultService was not found");
+
             _filterService = _engine.Services.GetService<ITestFilterService>();
+            Guard.OperationValid(_filterService != null, "Internal Error: TestFilterService was not found");
+
             _extensionService = _engine.Services.GetService<IExtensionService>();
+            Guard.OperationValid(_extensionService != null, "Internal Error: ExtensionService was not found");
+
+            var extensionPath = Environment.GetEnvironmentVariable(NUNIT_EXTENSION_DIRECTORIES);
+            if (!string.IsNullOrEmpty(extensionPath))
+                foreach (string extensionDirectory in extensionPath.Split(new[] { Path.PathSeparator }, StringSplitOptions.RemoveEmptyEntries))
+                    _extensionService.FindExtensions(extensionDirectory);
 
-            // TODO: Exit with error if any of the services are not found
+            foreach (string extensionDirectory in _options.ExtensionDirectories)
+               _extensionService.FindExtensions(extensionDirectory);
+
+            _workDirectory = options.WorkDirectory ?? Directory.GetCurrentDirectory();
+            if (!Directory.Exists(_workDirectory))
```

Comment:
nitpick: I'm unsure if it was a stylistic choice, but I think the `if` here isn't needed. I believe `Directory.CreateDirectory()` will instead do nothing if the directory already exists.

Reply:
Not actually new, but moved from elsewhere. I don't think it was needed even under .NET Framework 1.1.

Reply:
I changed it a bit but left the test because I really can't bear to call `Directory.CreateDirectory` on the current directory. :-)

Reply:
That's a good point about calling it on the current directory! 😃 I hadn't considered that. Thanks for having updated this.